cardinal/engine_test.go

```diff
@@ -344,6 +346,10 @@ func TestTransactionsSentToRouterAfterTick(t *testing.T) {
 	tx := &sign.Transaction{PersonaTag: "ty"}
 	_, txHash := world.AddEVMTransaction(fooMessage.ID(), msg, tx, evmTxHash)
 
+	fakeIterator := mocks.NewFakeIterator(nil)
+
+	rtr.EXPECT().TransactionIterator().Return(fakeIterator).Times(1)
+
```

Comment:
i see this is added because the only requirement for running RecoverFromChain is if theres a router or not. maybe this should just be if the world is in production?